sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1278,47 +1277,33 @@ static int get_slave_info(const struct snd_soc_acpi_link_adr *adr_link,
 	if (!is_power_of_2(adr_link->mask))
 		return -EINVAL;
 
-	cpu_dai_id[index++] = ffs(adr_link->mask) - 1;
 	if (!adr_d->endpoints->aggregated || no_aggregation) {
+		cpu_dai_id[0] = ffs(adr_link->mask) - 1;
 		*cpu_dai_num = 1;
 		*codec_num = 1;
 		*group_id = 0;
 		return 0;
 	}
 
+	*codec_num = 0;
 	*group_id = adr_d->endpoints->group_id;
 
 	/* Count endpoints with the same group_id in the adr_link */
-	*codec_num = 0;
-	for (i = 0; i < adr_link->num_adr; i++) {
-		if (adr_link->adr_d[i].endpoints->aggregated &&
-		    adr_link->adr_d[i].endpoints->group_id == *group_id)
-			(*codec_num)++;
-	}
-
-	/* gather other link ID of slaves in the same group */
-	for (adr_next = adr_link + 1; adr_next && adr_next->num_adr;
-		adr_next++) {
-		const struct snd_soc_acpi_endpoint *endpoint;
-
-		endpoint = adr_next->adr_d->endpoints;
-		if (!endpoint->aggregated ||
-		    endpoint->group_id != *group_id)
-			continue;
```

Comment:
Removing the test doesn't look right to me. We only want to add cpu_dai (adr_link) with the endpoint in the same group.
`cpu_dai_id[index++] = ffs(adr_link->mask) - 1;` in line 1302 will add all cpu_dai after the adr_link.
Take tgl_3_in_1_sdca[] for example,
```
static const struct snd_soc_acpi_link_adr tgl_3_in_1_sdca[] = {
        {
                .mask = BIT(0),
                .num_adr = ARRAY_SIZE(rt711_sdca_0_adr),
                .adr_d = rt711_sdca_0_adr,
        },
        {
                .mask = BIT(1),
                .num_adr = ARRAY_SIZE(rt1316_1_group1_adr),
                .adr_d = rt1316_1_group1_adr,
        },
        {
                .mask = BIT(2),
                .num_adr = ARRAY_SIZE(rt1316_2_group1_adr),
                .adr_d = rt1316_2_group1_adr,
        },
        {
                .mask = BIT(3),
                .num_adr = ARRAY_SIZE(rt714_3_adr),
                .adr_d = rt714_3_adr,
        },
        {}
};
```
the 4th adr_link (rt714_3) will be handled, and cpu_dai_id index will be 3 when `get_slave_info` is called for the aggregated rt1316

Reply:
Yeah I think you are correct here, good spot. I will rework.